opsdroid/connector/webexteams/__init__.py

```diff
@@ -0,0 +1,107 @@
+"""A connector for Webex Teams."""
+import json
+import logging
+import uuid
+
+import aiohttp
+
+from ciscosparkapi import CiscoSparkAPI
+
+from opsdroid.connector import Connector
+from opsdroid.message import Message
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConnectorWebexTeams(Connector):
+    """A connector for Webex Teams."""
+
+    def __init__(self, config):
+        """Create a connector."""
+        _LOGGER.debug("Starting webex teams connector")
+        self.name = "webexteams"
+        self.config = config
+        self.opsdroid = None
+        self.default_room = None
+        self.bot_name = config.get("bot-name", "opsdroid")
+        self.bot_spark_id = None
+        self.secret = uuid.uuid4().hex
+        self.people = {}
+
+    async def connect(self, opsdroid):
+        """Connect to the chat service."""
+        self.opsdroid = opsdroid
+        try:
+            self.api = CiscoSparkAPI(access_token=self.config["access-token"])
+        except KeyError:
+            _LOGGER.error("Must set accesst-token for webex teams connector!")
+            return
+
+        await self.clean_up_webhooks()
+        await self.subscribe_to_rooms()
+        await self.set_own_id()
+
+    async def webexteams_message_handler(self, request):
+        """Handle webhooks from the Cisco api."""
+        _LOGGER.debug("Handling message from Webex Teams")
+        req_data = await request.json()
+
+        _LOGGER.debug(req_data)
+
+        msg = self.api.messages.get(req_data["data"]["id"])
+
+        if req_data["data"]["personId"] != self.bot_spark_id:
+            person = await self.get_person(req_data["data"]["personId"])
+
+            try:
+                message = Message(
+                    msg.text,
+                    person.displayName,
+                    {"id": msg.roomId, "type": msg.roomType},
+                    self,
+                )
+                await self.opsdroid.parse(message)
+            except KeyError as error:
+                _LOGGER.error(error)
+
+        return aiohttp.web.Response(text=json.dumps("Received"), status=201)
+
+    async def clean_up_webhooks(self):
+        """Remove all existing webhooks."""
+        for webhook in self.api.webhooks.list():
+            self.api.webhooks.delete(webhook.id)
+
+    async def subscribe_to_rooms(self):
+        """Create webhooks for all rooms."""
+        _LOGGER.debug("Creating Webex Teams webhook")
+        webhook_endpoint = "/connector/ciscospark"
```

Comment:
We probably need to rename the endpoint too

```suggestion
        webhook_endpoint = "/connector/webexteams"
```